crates/router/src/core/payment_link/payment_link_initiate/payment_link.js

```diff
@@ -617,7 +617,7 @@ function renderDynamicMerchantDetails(paymentDetails) {
 }
 
 function appendMerchantDetails(paymentDetails, merchantDynamicDetails) {
-  if (Object.keys(paymentDetails.transaction_details).length === 0) {
+  if (paymentDetails.transaction_details === null || paymentDetails.transaction_details === undefined || paymentDetails.transaction_details === "") {
```

Comment:
can we add a check for the required type rather than checking for all the combinations of invalid types?

```suggestion
  if (
    !(
      typeof paymentDetails.transaction_details === "object" &&
      !Array.isArray(paymentDetails.transaction_details) &&
      Object.keys(paymentDetails.transaction_details).length > 0
    )
  ) {
```